src/execution_plan/execution_plan_build/execution_plan_util.c

```
@@ -305,24 +331,80 @@
+
+	// collect upwards using child pointers
+	while(s < e) {
+		OpBase *current = ops[s++];
+
+		// try adding current children
+		for(uint i = 0; i < OpBase_ChildCount(current); i++) {
+			OpBase *child = OpBase_GetChild(current, i);
+
+			// add child only if it belong to given plan
+			if(child->plan == plan) {
+				ops = rm_realloc(ops, sizeof(OpBase*) * (e + 1));
```

cppcheck (2.10-2)

[error] Common realloc mistake

(memleakOnRealloc)